kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -88,6 +92,8 @@ public static TransactionRebaseState resolveConflicts(
 
     public TransactionRebaseState resolveConflicts(Engine engine) throws ConcurrentWriteException {
         List<FileStatus> winningCommits = getWinningCommitFiles(engine);
+        AtomicReference<Optional<CommitInfo>> winningCommitInfoOpt = new AtomicReference<>(
```

Comment:
Any reason why this needs to an `AtomicReference`?

Reply:
Because of this error:
<img width="721" alt="image" src="https://github.com/delta-io/delta/assets/55441375/fecb20f7-8e0c-4418-b589-a877f06b18d8">